### PR TITLE
move options interface definition to separated interface and export it

### PR DESCRIPTION
**Pull request by Karpengold**

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -22,9 +22,7 @@ export interface Range {
   size: number
 }
 
-declare function defaultRangeExtractor(range: Range): number[]
-
-declare function useVirtual<T>(options: {
+export interface ReactVirtualOptions {
```
Comment (piecyk) [inline types/index.d.ts:25]: Missing generic param `T`
```suggestion
export interface ReactVirtualOptions<T> {
```

File: types/index.d.ts [hunk 2]
```diff
@@ -47,7 +45,11 @@ declare function useVirtual<T>(options: {
   onScrollElement?: React.RefObject<HTMLElement>
   scrollOffsetFn?: (event?: Event) => number
   rangeExtractor?: (range: Range) => number[]
-}): {
+}
+
+declare function defaultRangeExtractor(range: Range): number[]
+
+declare function useVirtual<T>(options: ReactVirtualOptions): {
```
Comment (piecyk) [inline types/index.d.ts:52]: Pass generic param to `Options`

```suggestion
declare function useVirtual<T>(options: Options<T>): {
```

File: types/index.d.ts (hunk 1 above)
Comment (piecyk) [reply types/index.d.ts:25]: Also imho would drop the `ReactVirtual` prefix 

```suggestion
export interface Options<T> {
```

User can always rename it when importing, `import { useVirtual, Options as ReactVirtualOptions } from 'react-virtual'`

**Review by piecyk — CHANGES_REQUESTED** [review]

File: types/index.d.ts (hunk 1 above)
Comment (Karpengold) [reply types/index.d.ts]: yeah, don't know how I missed generic param :) vs code even highlight it!

File: types/index.d.ts (hunk 2 above)
Comment (Karpengold) [reply types/index.d.ts]: updated!

**Review by piecyk — APPROVED** [review]

**Pull request merged**